### PR TITLE
CI: Refactor typedoc/storybook

### PR DESCRIPTION
**Pull request by 4ydan**
Removes redundant builds.

## Basics

- [ ] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -105,7 +105,7 @@ generate-type-doc:  ## Generate typedoc.
 
 .PHONY: generate-api-types
 generate-api-types:  ## Generate api-types.
-	cd frontend && npm run generate-api-types
+	-cd frontend && npm run generate-api-types
```
Comment (4ydan) [inline Makefile:108]: There is a workaround here, I m not sure how to solve this yet. When using the makefile in CI now, we perform generate-api-types for the storybook, but we dont have typeshare installed in that docker, so in CI they are moved over stages, but locally we want to just create the types fresh, maybe just remove type generation.

Comment (markus2330) [reply Makefile:108]: Why not simply depend on the generated file? Then it should not matter if it was actually generated (local builds) or copied from other stages (CI).

Comment (4ydan) [reply Makefile]: Totally forgot about that.
I didnt manage to do it as [pre-requisite](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html), so I added this if.

I basically tried this:
```
.PHONY: generate-api-types
generate-api-types:  ./frontend/src/bindings/definitions.ts  ## Generate api-types.
	my-code
```

Comment (markus2330) [reply Makefile]: Do we maybe have a misunderstanding about the term pre-requisite? With pre-requisite I meant text written by us in the documentation that a user must have install packages (or use our Docker file) before using our Makefiles.

Comment (4ydan) [reply Makefile]: > Why not simply depend on the generated file?

With pre-requisite I mean the file dependance you mentioned.

File: Makefile [hunk 2]
```diff
@@ -69,7 +69,7 @@ build-mdbook:  ## Build mdbook.
 	mdbook build
 
 .PHONY: build-storybook
-build-storybook: install generate-api-types  ## Build storybook.
+build-storybook:  ## Build storybook.
```
Comment (markus2330) [inline Makefile:72]: I don't really get this change, both dependencies are needed? And furthermore `npm run doc`, is needed, too.

File: Makefile [hunk 3]
```diff
@@ -105,7 +105,9 @@ generate-type-doc:  ## Generate typedoc.
 
 .PHONY: generate-api-types
 generate-api-types:  ## Generate api-types.
-	cd frontend && npm run generate-api-types
+	@if [ ! -f "./frontend/src/bindings/definitions.ts" ]; then \
```
Comment (markus2330) [inline Makefile:108]: Why not depend on /frontend/src/bindings/definitions.ts?

File: ci/Jenkinsfile [hunk 4]
```diff
@@ -165,13 +165,13 @@ def testAndBuildFrontend() {
                     stash includes: 'frontend/dist/**/*', name: 'frontend'
                 }
 
-                stage('typedoc') {
-                    sh 'cd frontend && npm run doc'
+                stage('build-typedoc') {
+                    sh 'make generate-type-doc'
                     stash includes: 'frontend/src/generated/docs/', name: 'typedoc'
                 }
```
Comment (markus2330) [inline ci/Jenkinsfile:171]: I don't think this stage or artifact is needed. Afaik `npm run doc` is only needed before storybook to build storybook correctly.

**Review by markus2330 — COMMENTED** [review]
Some questions. Imho it is okay to merge this and you continue your work in #681.

**Comment by markus2330** [comment]
I removed "please merge" as I am not sure if everything is as you wanted. Please confirm by adding "please merge" again.

**Comment by 4ydan** [comment]
It's currently not very easy to use the make commands in CI, because for example `make doc-storybook` for example does `make install-types` which does `make install` which does `rustup component add rustfmt clippy` which requires rust which we dont have in the frontend dockerfile.

So we either close this PR and dont try to bring more make commands into CI or we go back to creating one big dockerfile which contains everything.

**Comment by markus2330** [comment]
Thx for trying. Let us forget this for now.

**Pull request closed without merging**